### PR TITLE
Solc 0.6.0

### PR DESCRIPTION
**Pull request by MrChico**
draft pr for visibility,
Non-exhaustive list of things that need fixing in order to support `0.6.0` are:
-  Upgrade ds-test to follow the breaking changes convensions
-  Fix `Solidity.hs` to accomodate the new srcmap format

### CODE REVIEW

**Comment by MrChico** [comment]
The main hurdle to allowing for solc-0.6 has been the breaking change in the srcmap format, which this PR supports. Together with https://github.com/dapphub/ds-test/pull/14, the basic `dapp` workflow now works with `solc-0.6`, even if a lot of ds packages still need updating.

I suggest we don't let updating the rest of ds-* block this though

File: src/hevm/src/EVM/Solidity.hs [hunk 1]
```diff
@@ -139,44 +140,49 @@ makeLenses ''Method
...
   where
-    digits ds = digits' (0 :: Int) (0 :: Int) ds
-    digits' !x _ []      = x
-    digits' !x !n (d:ds) = digits' (x + d * 10 ^ n) (n + 1) ds
-
     done (xs, s, p) = let (xs', _, _) = go ';' (xs, s, p) in xs'
 
-    go ':' (xs, F1 [] _, p@(SM a _ _ _))       = (xs, F2 a [] 1, p)
-    go ':' (xs, F1 ds k, p)                    = (xs, F2 (k * digits ds) [] 1, p)
+    go :: Char -> (Seq SrcMap, SrcMapParseState, SrcMap) -> (Seq SrcMap, SrcMapParseState, SrcMap)
```
Comment (MrChico) [inline src/hevm/src/EVM/Solidity.hs]: Good luck reviewing this!

File: src/dapp/CHANGELOG.md [hunk 2]
```diff
@@ -8,9 +8,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Integration with `hevm`s new property based testing functionality. Any test with nonzero arguments will be interpreted as a property test whose arguments are randomly generated and run `--fuzz-runs` number of times.
 - Support for solc 0.5.16
+- Support for solc 0.6.0
 
 ## [0.26.0]
 ### Added
+=======
```
Comment (asymmetric) [inline src/dapp/CHANGELOG.md]: Wut?

Comment (MrChico) [reply src/dapp/CHANGELOG.md]: oops

**Comment by asymmetric** [comment]
I'm fine with merging this if you've tested it.

**Pull request merged**